CMakeLists.txt

```diff
@@ -46,7 +46,8 @@ add_subdirectory(vendor)
 add_subdirectory(src)
 
 if (${PROJECT_NAME}_BUILD_TESTING)
-    add_subdirectory(test)
+    add_subdirectory(test/unit_tests)
```

Comment:
One very minor change here: This subdirectory should be added from the test/CMakeLists.txt.